frontend/src/App.test.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { shallow, configure } from "enzyme";
+import Adapter from 'enzyme-adapter-react-16';
+import { enableFetchMocks } from 'jest-fetch-mock';
+enableFetchMocks();
+import App from './App';
+
+configure({ adapter: new Adapter() });
+
+const LOGIN_URL = "/_ah/login?continue=%2F";
+const LOGOUT_URL = "/_ah/logout?continue=%2F";
+const LOGIN_MESSAGE = "logged in\n/_ah/logout?continue=%2F";
+const LOGOUT_MESSAGE = "stranger\n/_ah/login?continue=%2F";
+
+describe("getLoginStatus", () => {
+
+  beforeEach(() => {
+    fetch.resetMocks();
+  });
+
+  it("sets logged in state when user is logged in", (done) => {
```

Comment:
what about the case where initially user is logged in but then clicks log out.

initial state: `component.setState({ isLoggedIn: true, logURL: ""});`

Reply:
I should rephrase the tests. This test is setting the logged in state when the user *logs in*. The next test is setting the logged out state when the user *logs out*. does that clear it up?